### ISSUE TITLE
Local installation: Unknown MySQL server host

### ISSUE THREAD

**Issue by dominikbraun** [shopware/shopware#51]
Then I run `./psh.phar install` as suggested in the [guide for local installation](https://docs.shopware.com/en/shopware-platform-dev-en/getting-started/installation-guide#setting-up-shopware), an error occurs:

```
ERROR 2005 (HY000): Unknown MySQL server host 'mysql' (2)
```

This is due to the `-h 'mysql'` option for the `mysql` command at step 3/28.

I fixed the error by registering `127.0.0.1 mysql` in `/etc/hosts`. Maybe this information should be added to the installation guide?

**Comment by shyim** [comment]
I guess you did a local installation. So you had to run ./bin/setup first. Where you can define the credentials

**Comment by dominikbraun** [comment]
Yes, but the database host in the setup defaults to `localhost`, not `mysql`.

**Comment by shyim** [comment]
Thats the important part https://github.com/shopware/development/blob/master/.psh.yaml.dist#L23

**Comment by DanielRuf** [comment]
Well, but this does not work with the Docker containers where `app_mysql` is built.
Because if we ssh into it, the composer taks fails due to the missing CA certificate file.

```
(1/28) Starting
> php dev-ops/generate_ssl.php
	Private/Public key already exists. Skipping
(2/28) Starting
> composer install --no-interaction --optimize-autoloader --no-suggest --no-scripts
	Unable to locate a valid CA certificate file. You must set a valid 'cafile' option.
	A valid CA certificate file is required for SSL/TLS protection.
	You can disable this error, at your own risk, by setting the 'disable-tls' option to true.
	
	                                                             
	  [Composer\Downloader\TransportException]                   
	  The configured cafile was not valid or could not be read.
```

So in both cases the default setup does not directly work.
I guess we need at least some changes for the Docker images. At least when we want to follow the steps from the repository.

https://github.com/shopware/platform#quickstart--installation

**Comment by wanicode** [comment]
`(1/28) Starting
> php dev-ops/generate_ssl.php
        Private/Public key already exists. Skipping
(2/28) Starting
> composer install --no-interaction --optimize-autoloader --no-suggest --no-scripts
        Loading composer repositories with package information
        Installing dependencies (including require-dev) from lock file
        Nothing to install or update
        Generating optimized autoload files

(3/28) Starting
> mysql -u 'app' -p'app' -h 'mysql' --port='3306' -e "DROP DATABASE IF EXISTS \`shopware\`"
        ERROR 2005 (HY000): Unknown MySQL server host 'mysql' (-2)
`

I get this when i try to do `./psh.phar install` after going into the docker-container.

**Comment by keulinho** [comment]
To sum it up:
The default value  for the DB host is `mysql`, by running `./bin/setup` you can override this default.
In the setup script we use `localhost` as the default, because when you run the setup script you problably want to install it locally, note that this will also just override the `mysql` for the DB Host.
You can find more information about how this overriding of config values works [here](https://github.com/shopware/platform/issues/65#issuecomment-498946980).

@DanielRuf i haven't seen that error message. 
What steps did you take?
Did run the `./bin/setup` script before starting the docker container? 
You don't need to because the default config values we provide directly work with our docker-setup.

Lastly did you run the `./psh.phar install` command inside the container or on your local system?
Maybe the installation guide in the readme can be misleading here, but you need to run the `install` command inside your container.

**Comment by DanielRuf** [comment]
> Did run the `./bin/setup` script before starting the docker container?

If the psh.phar install task does this, yes:
https://github.com/shopware/platform#quickstart--installation

**Comment by DanielRuf** [comment]
> Lastly did you run the `./psh.phar install` command inside the container or on your local system?

Inside. I did not run any other `.bin/setup` as this is not mentioned there.
If this is needed it should be added to the docs.

**Comment by DanielRuf** [comment]
> `app_mysql` is built.

app_mysql !== mysql ;-)
https://github.com/shopware/development/search?q=app_mysql&unscoped_q=app_mysql

Not sure but it did not find the mysql instance by default.

**Comment by keulinho** [comment]
You can see in the docker-compose file the app_mysql container is linked to the app container with the name `mysql`, so the DB host seen from the app container is `mysql`. https://github.com/shopware/development/blob/18ca00ac721fbc271b60645a3fa9304810820567/docker-compose.yml#L7

> >Did run the ./bin/setup script before starting the docker container?
> 
> If the psh.phar install task does this, yes:

You don't have to run the `./psh.phar install` command before starting the container.
If you use the docker setup all psh commands you may need to run on your local machine are prefixed with `docker:`.
E.g.
`./psh.phar docker:start`
`./psh.phar docker:ssh`
`./psh.phar docker:stop`

**Comment by DanielRuf** [comment]
> `./psh.phar install`

I tried this in the container, was the same result.
Not 100% sure as it was failing with the given steps in the docs - there might be something unclear.

**Comment by janbuecker** [comment]
I guess your problem has been fixed? Feel free to reopen this issue if it's not.

**Comment by DanielRuf** [comment]
> I tried this in the container, was the same result.

So far it was not fixed =)
I did not invest more time after this.

**Comment by go2muneeb** [comment]
from the last 6 hours its installing .psh.phar  what should I do now plz help. 

TASK [shopware-dev : disable 000-default.conf] *********************************
changed: [default]
TASK [shopware-dev : execute psh.phar install] *********************************
![psh](https://user-images.githubusercontent.com/60916158/78462259-c737c580-76e9-11ea-8b5c-6ef91d1a5fba.PNG)

**Comment by DanielRuf** [comment]
Please open a new issue @go2muneeb.

**Comment by shinuskn** [comment]
First run **./psh.phar docker:stop**
Then run the following commands
**./psh.phar docker:start
./psh.phar docker:ssh
./psh.phar install**

**Comment by nroehling** [comment]
> First run **./psh.phar docker:stop**
> Then run the following commands
> **./psh.phar docker:start ./psh.phar docker:ssh ./psh.phar install**

that worked for me!
